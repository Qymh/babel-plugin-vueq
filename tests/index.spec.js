const babel = require('@babel/core');
const plugin = require('../lib/index');

it('works', () => {
  const template = `
      import {QButton} from 'vueq-mobile-ui'
    `;
  const { code } = babel.transform(template, { plugins: [plugin] });
  expect(code).toMatchSnapshot();
});

it('match vueq-mobile-ui', () => {
  const template = `
      import {QButton} from 'vueq-mobile-ui'
    `;
  const { code } = babel.transform(template, { plugins: [plugin] });
  expect(code).toMatch(/vueq-mobile-ui\/dist\/packages\/QButton\.js/);
  expect(code).toMatch(/vueq-mobile-ui\/dist\/css\/QButton\.css/);
});

it('not match vueq-mobile-ui', () => {
  const template = `
    import {QButton} from 'othermobule'
  `;
  const { code } = babel.transform(template, { plugins: [plugin] });
  expect(code).not.toMatch(/vueq-mobile-ui\/dist\/packages\/QButton\.js/);
});

it('no chunks', () => {
  const template = `
    import {} from 'vueq-mobile-ui'
  `;
  const { code } = babel.transform(template, { plugins: [plugin] });
  expect(code).not.toMatch(/\{/);
});

it('only default', () => {
  const template = `
    import vueq from 'vueq-mobile-ui'
  `;
  const { code } = babel.transform(template, { plugins: [plugin] });
  expect(code).toBe(`import vueq from 'vueq-mobile-ui';`);
});

it('defualt with specifier', () => {
  const template = `
    import vueq,{QButton} from 'vueq-mobile-ui'
  `;
  const { code } = babel.transform(template, { plugins: [plugin] });
  expect(code).toMatch(/vueq-mobile-ui\/dist\/packages\/QButton\.js/);
  expect(code).toMatch(/vueq-mobile-ui\/dist\/css\/QButton\.css/);
});

it('no css', () => {
  const template = `
    import {QCol} from 'vueq-mobile-ui'
  `;

  const { code } = babel.transform(template, { plugins: [plugin] });
  expect(code).toMatch(/vueq-mobile-ui\/dist\/packages\/QCol\.js/);
  expect(code).not.toMatch(/vueq-mobile-ui\/dist\/css\/QCol\.css/);
});
