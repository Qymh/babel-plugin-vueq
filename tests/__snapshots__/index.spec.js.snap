// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"import QButton from \\"vueq-mobile-ui/dist/packages/QButton.js\\";
import \\"vueq-mobile-ui/dist/css/QButton.css\\";"
`;
